### ISSUE TITLE
ParallelInference does not support masking out of the box

### ISSUE THREAD

**Issue by mboyanov** [deeplearning4j/deeplearning4j#4489]
#### Issue Description
I'm importing a Keras model which contains an Embedding layer trained with mask_zero=True. Since dl4j and Keras implement masking differently this produces the following warning:
```
2018-01-09 12:43:32,282 WARN  [org.deeplearning4j.nn.modelimport.keras.layers.embeddings.KerasEmbedding] Masking in keras and DL4J work differently. We do not support mask_zero flagon Embedding layers. If you want to have this behaviour for your imported modelin DL4J, apply masking as a pre-processing step to your input.See https://deeplearning4j.org/usingrnns#masking for more on this.
```

If the `ComputationGraph` is used directly, this is fine, because it is possible to compute the mask arrays and call `.setLayerMaskArrays` prior to predicting/classifying or whatever.
However, I'm using `ParallelInference`, which does not have an API to supply the masks. 

## Model
The relevant part of the model in question is a standard input layer ->embeddings layer ->lstm layer .
The input has shape (batch_size, 1, 40) where the pad values to be masked are zeroes, e.g. `[0 0 0 .. 35 41 23]`.

## Expected:
ParallelInference should support input masking 
OR
there should be some kind of workaround. Maybe some kind of ZeroMaskingPreProcesser?

## Actual:
ParallelInference does not support input masking.

#### Version Information

Please indicate relevant versions, including, if relevant:

* Deeplearning4j version 0.9.2-SNAPSHOT

**Comment by AlexDBlack** [comment]
Input mask support was added here: https://github.com/deeplearning4j/deeplearning4j/pull/4836
